draft-irtf-cfrg-opaque.md

```diff
@@ -638,21 +647,22 @@ Output:
 Steps:
 1. y = Finalize(password, blind, response.data)
 2. envelope_nonce = random(32)
-3. prk = Extract(envelope_nonce, Harden(y, params))
+3. prk = Extract(salt=0, Harden(y, params))
```

Comment:
```suggestion
3. prk = Extract("", Harden(y, params))
```

Reply:
Cool, also changed the salt=0 instance to "" in the "OPAQUE-3DH Key Schedule" section

Reply:
Actually, it is better to set the salt to 0. It does not have to be described as salt=0 but just 0 (instead of ""). In principle, any fixed salt value can do, but I prefer to see 0 being used as the default "fixed value". In particular, 0 is used in TLS 1.3 key derivation when a random value for salt is not available and here we want to stay as close as possible to TLS 1.3 key derivation.

Reply:
For what it's worth, "" is the notation used in the TLS zero-length strings, which is what we want here, I think.

Reply:
What is the meaning of 0 in the following TLS 1.3 figure. Is it the value 0 or a 0-length string?
             0
             |
             v
   PSK ->  HKDF-Extract = Early Secret

Reply:
Actually I see the answer: "0" indicates a string of Hash.length bytes set to zero.

We should specify the same here.

Reply:
That's a zero-length string (or NULL, or the equivalent of providing no salt at all), but I was referring to "" used elsewhere where context strings.

Reply:
> Actually I see the answer: "0" indicates a string of Hash.length bytes set to zero.

I'm not sure this is best, since the KDF might not be HKDF (and therefore have an implied hash function underneath). I think nil/""/0/whatever are equivalent, and "" seems to be most clear (to me) and consistent with 8446.

Reply:
Good point. If we are treating the KDF as a general Extract-then-Expand primitive then we cannot say "a string of Hash.length bytes set to zero" but we also cannot say the empty string as it may not make sense in some other KDF. If you want to be generic, then you should refer to this salt value as a "constant" (as opposed to a random one), maybe call it const_salt and set it to "a string of Hash.length bytes set to zero" in the case of HKDF.